client/posts/embed.ts

```diff
@@ -60,29 +67,60 @@ function formatNoEmbed(type: provider): (s: string) => string {
 	}
 }
 
+// fetcher for the HookTube provider
+function fetchHookTube(): (el: Element) => Promise<void> {
+	return async (el: Element) => {
+		// Use a CORS proxy to work around javascript's cross-domain restrictions
+		const url = "https://hooktube.com/embed/" +
+			el.getAttribute("href").split("watch?v=").pop().split("/embed/").pop() + "?autoplay=false",
+			[data, err] = await fetchJSON<any>("https://cors-proxy.htmldriven.com/?url=" + url)
+
+		if (err) {
+			el.textContent = format(err, provider.HookTube)
+			el.classList.add("errored")
+			console.error(err)
+			return
+		}
+
+		if (data.error) {
+			el.textContent = format(data.error, provider.HookTube)
+			el.classList.add("errored")
+			return
+		}
+
+		el.textContent = format(new DOMParser().parseFromString(data.body, "text/html").
+			querySelectorAll('title')[0].innerText, provider.HookTube)
+		el.setAttribute("data-html", encodeURIComponent("<iframe width=\"480\" height=\"270\" src=\"" +
+			url + "\" frameborder=\"0\"</iframe>"))
+	}
+}
+
 // fetcher for the noembed.com meta-provider
 function fetchNoEmbed(type: provider): (el: Element) => Promise<void> {
 	return async (el: Element) => {
```

Comment:
I'm fixing it myself, but just FYI, there really is no point in this function factory, since you don't use any context parameters. Just a simple async function is enough.